### ISSUE TITLE
Azurerm_kusto_cluster deployment not honoring zones in code causing data loss

### ISSUE THREAD

**Issue by lulujrush** [hashicorp/terraform-provider-azurerm#17461]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.1.8

### AzureRM Provider Version

3.11.0

### Affected Resource(s)/Data Source(s)

zurerm_kusto_cluster

### Terraform Configuration Files

```hcl
resource "azurerm_resource_group" "example" {
  name     = "my-kusto-cluster-rg"
  location = "eastus2"
}

resource "azurerm_kusto_cluster" "data_lake" {
  name                = "kustocluster"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  engine              = "V3"
  auto_stop_enabled   = true
  zones               = [1,2]

  sku {
    name     = "Standard_D11_v2"
    capacity = "2"
  }

  optimized_auto_scale {
    minimum_instances = 2
    maximum_instances = 3
  }

  tags = {
    environment                       = var.env
    terraform                         = "true"
  }
}
```

### Debug Output/Panic Output

```shell
n/a
```

### Expected Behaviour

In my code I set the ADX Zones to [1,2] expected behavior is that ADX is deployed to Zones [1,2]

### Actual Behaviour

ADX is being deployed into [1,3]

### Steps to Reproduce

New deployment with [1,2] in my code, `tf apply` shows that it will be going to [1,2]

<img width="540" alt="Screen Shot 2022-06-30 at 08 32 31" src="https://user-images.githubusercontent.com/101839303/176728903-4d55cc9b-0a06-479e-a0be-1a298276b152.png">

but in reality when deployed it goes to [1,3]
![Screen Shot 2022-06-30 at 08 36 10](https://user-images.githubusercontent.com/101839303/176728971-9d56cd97-e7fa-44fc-b300-d8638f21c55f.png)

json view of console
![Screen Shot 2022-06-30 at 08 36 47](https://user-images.githubusercontent.com/101839303/176729008-cca395d0-d1dd-402d-8ffe-e62a7577eb0e.png)

then if I do another `tf apply` we see that it also sees the deployed item is NOT what it thought was going to be and will destroy/replace.
<img width="640" alt="Screen Shot 2022-06-30 at 08 43 41" src="https://user-images.githubusercontent.com/101839303/176729096-f66de147-c527-4e55-8850-2a548d0692f0.png">

so I let that run and then in Azure it STILL shows [1,3] after deployments complete.
![Screen Shot 2022-06-30 at 09 10 45](https://user-images.githubusercontent.com/101839303/176729126-694d23d5-f716-428a-9531-f412d3c19e64.png)

### References

I found a similar issue but not quite what I am experiencing so I decided to make this new issue. Also that one is over a year old with no traction on it. 

https://github.com/hashicorp/terraform-provider-azurerm/issues/12504

**Comment by lulujrush** [comment]
Forgot to add and just updated title that this is causing data loss as the underlying databases on the cluster are also being destroyed.

**Comment by magodo** [comment]
@lulujrush This seems to be an API bug, the kusto service silently changes the `zones` during creation, see https://github.com/Azure/azure-rest-api-specs/issues/19645

**Comment by liuwuliuyun** [comment]
I have contacted kusto team for this bug before and they are aware of this. Here is what happened, when you run terraform apply for the first time, they will put your resource to the correct zone. 
However, when you get the status using terraform refresh, the GET api return the  wrong zone and cause diff in states.

**Comment by liuwuliuyun** [comment]
I checked this last night and the bug still exists. So I will try to reach out to them and see when they can fix the bug.

**Comment by lulujrush** [comment]
@magodo and @liuwuliuyun  thank you for both helping us dig into this issue, it has been quite frustrating as you can imagine. :D 

@liuwuliuyun in my testing even an initial deployment does not respect the zone choices. 

Should we leave this issue open until the linked issue from @magodo has been resolved and then I can validate fix and close this?

**Comment by tombuildsstuff** [comment]
@liuwuliuyun whilst we're waiting on the Service Team to fix this, we should also look to split the Create and Update method here (to use `d.HasChanges(..)` to conditionally update values) as well - which'd allow folks to workaround this using `ignore_changes` whilst the Service Team fixes the API issue.

**Comment by liuwuliuyun** [comment]
GOROOT=D:\Program Files\Go #gosetup
GOPATH=C:\Users\yunliu1\go #gosetup
"D:\Program Files\Go\bin\go.exe" test -c -o C:\Users\yunliu1\AppData\Local\Temp\GoLand\___TestAccKustoCluster_basic_in_github_com_hashicorp_terraform_provider_azurerm_internal_services_kusto.test.exe github.com/hashicorp/terraform-provider-azurerm/internal/services/kusto #gosetup
"D:\Program Files\Go\bin\go.exe" tool test2json -t C:\Users\yunliu1\AppData\Local\Temp\GoLand\___TestAccKustoCluster_basic_in_github_com_hashicorp_terraform_provider_azurerm_internal_services_kusto.test.exe -test.v -test.paniconexit0 -test.run ^\QTestAccKustoCluster_basic\E$ #gosetup
=== RUN   TestAccKustoCluster_basic
=== PAUSE TestAccKustoCluster_basic
=== CONT  TestAccKustoCluster_basic
--- PASS: TestAccKustoCluster_basic (1254.71s)
PASS


Process finished with the exit code 0

**Comment by lulujrush** [comment]
Hi @liuwuliuyun - I actually noticed yesterday myself when I was working on our ADX stack that this was no longer happening.. Thank you everyone involved in helping us figure out this bug. 

I have been pushing us towards Terraform with some hesitation so it is great to be able to show how the community supports each other to my team. :D 

I will go ahead and close this issue with this comment.
